thunx-spring/src/main/java/eu/xenit/contentcloud/thunx/spring/data/rest/AbacRepositoryInvokerAdapter.java

```diff
@@ -139,8 +141,59 @@ private String toAlias(Class<?> subjectType) {
     }
 
     private BooleanExpression idExpr(Object id, PathBuilder entityPath) {
-        Field idField = BeanUtils.findFieldWithAnnotation(domainType, Id.class);
+        Field idField = DomainObjectUtils.getIdField(domainType);
         PathBuilder idPath = entityPath.get(idField.getName(), id.getClass());
         return idPath.eq(Expressions.constant(id));
     }
+
+
+    static class DomainObjectUtils {
+
+        private static final boolean JAVAX_PERSISTENCE_PRESENT = ClassUtils.isPresent(
+                "javax.persistence.Id", DomainObjectUtils.class.getClassLoader());
+
+        static final Field getIdField(Class<?> domainClass) {
+
+            Optional<Field> javaxPersistenceId = JAVAX_PERSISTENCE_PRESENT
+                    ? DomainObjectUtils.findFieldWithAnnotation(domainClass, javax.persistence.Id.class)
+                    : Optional.empty();
+
+            return javaxPersistenceId
+                    .or(() -> DomainObjectUtils.findFieldWithAnnotation(domainClass, org.springframework.data.annotation.Id.class))
+                    .orElse(null);
+        }
+
+        private static Optional<Field> findFieldWithAnnotation(Class<?> domainObjClass,
+                                                                 Class<? extends Annotation> annotationClass)
```

Comment:
Bizarre indentation, not aligned with anything? It should just have the normal continuation indent of pressing tab twice, like you did for the `throws` line under this.